src/route/user-route.ts

```diff
@@ -0,0 +1,41 @@
+import { Router } from 'express';
+
+// eslint-disable-next-line import/no-cycle
+import { logger } from '../app';
+import { User } from '../entity/user';
+import { userToUserDTO } from '../dtos/user-dto';
+
+export const userRoute = Router();
+
+userRoute.get('/:id', async (req, res) => {
+    if (!req.params.id) {
+        logger.error(`Reequest parameter id is missing`);
+        res.status(400).send();
+        return;
+    }
+    const userID = req.params.id;
+    const user = await User.findOneBy({ id: userID });
+    if (!user) {
+        logger.error(`User with id ${req.params.id} not found`);
+        res.status(404).send();
+        return;
+    }
+    const userDto = userToUserDTO(user);
+    res.json(userDto);
+});
+
+userRoute.post('/', async (req, res) => {
+    const user = new User();
+    user.oidcId = req.body.oidcId;
+    user.provider = req.body.provider;
+    user.name = req.body.name;
+    user.email = req.body.email;
+    user.profile = req.body.profile;
```

Comment:
All this data should come from the IDP.
If there's code to verify that in the OIDC library then we can rely on it but we need a comment here to explain it.
Otherwise we need to validate all of these things before putting them in the database.
The name and eMail, which will later be used in HTML Responses are particularly sensitive.

Reply:
i.e. we need to ensure it's safe if injected by an untrusted attacker (and specify how much we trust the IDP and our OIDC client). Document (in a comment) what attribute checking has happened up to this point and then do whatever other checking we need in code.